x-pack/plugins/reporting/server/config/index.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TypeOf } from '@kbn/config-schema';
+import crypto from 'crypto';
+import { map } from 'rxjs/operators';
+import { PluginConfigDescriptor } from 'kibana/server';
+import { CoreSetup, Logger, PluginInitializerContext } from '../../../../../src/core/server';
+import { ConfigSchema, ConfigType } from './schema';
+
+export function createConfig$(core: CoreSetup, context: PluginInitializerContext, logger: Logger) {
+  return context.config.create<TypeOf<typeof ConfigSchema>>().pipe(
+    map(config => {
+      let encryptionKey = config.encryptionKey;
+      if (encryptionKey === undefined) {
+        logger.warn(
+          'Generating a random key for xpack.reporting.encryptionKey. To prevent sessions from being invalidated on ' +
+            'restart, please set xpack.reporting.encryptionKey in kibana.yml'
+        );
+        encryptionKey = crypto.randomBytes(16).toString('hex');
+      }
+
+      const { kibanaServer: reportingServer } = config;
+      const serverInfo = core.http.getServerInfo();
+
+      // kibanaServer.hostname, default to server.host
+      const kibanaServerHostname = reportingServer.hostname
+        ? reportingServer.hostname
+        : serverInfo.host;
+
+      // kibanaServer.port, default to server.port
+      const kibanaServerPort = reportingServer.port
+        ? reportingServer.port
+        : serverInfo.port; // prettier-ignore
+
+      // kibanaServer.protocol, default to server.protocol
+      const kibanaServerProtocol = reportingServer.protocol
+        ? reportingServer.protocol
+        : serverInfo.protocol;
+
+      return {
+        ...config,
+        encryptionKey,
+        kibanaServer: {
+          hostname: kibanaServerHostname,
+          port: kibanaServerPort,
+          protocol: kibanaServerProtocol,
+        },
+      };
+    })
+  );
+}
+
+export const config: PluginConfigDescriptor<ConfigType> = {
+  schema: ConfigSchema,
+  deprecations: ({ unusedFromRoot }) => [
+    unusedFromRoot('capture.concurrency'),
+    unusedFromRoot('capture.timeout'),
+    unusedFromRoot('capture.settleTime'),
+    unusedFromRoot('kibanaApp'),
```

Comment:
I've been trying to test this by adding these unused entries in my `kibana.yml`

My first attempt was to try it without any of these properties existing in the schema, and Kibana wouldn't start.

Once I added the properties to the schema, Kibana would start but there are no errors logged, as would be the case with the legacy config, which I tested by switching to master.

Basically, it seems like the `deprecations` function isn't doing anything.